[USH-2623] messaging subevent domain index and usage

snopoke opened the pull request:
Jira: https://dimagi-dev.atlassian.net/browse/USH-2623

## Technical Summary
Builds on https://github.com/dimagi/commcare-hq/pull/32339:

* adds the combined index (and remove the single field index on 'date_last_activity')
* updates usage of the table to use the new domain field

## Safety Assurance

### Safety story
The changes are made in code that is tested by unit tests. 

Prior to rolling this out there will be checks to ensure that the new column is fully populated.

### Automated test coverage
Existing tests

### QA Plan
None


### Migrations
<!-- Delete this section if the PR does not contain any migrations -->
<!-- https://commcare-hq.readthedocs.io/migrations_in_practice.html -->
- [x] The migrations in this code can be safely applied first independently of the code

The migration in this PR will be executed prior to deploy (after the PR is merged) to ensure that deploy does not block on the migration.

<!-- Please link to any past code changes that are coordinated with this migration -->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [ ] This PR can be reverted after deploy with no further considerations

If the changes are rolled back the DB state will be out of sync with code. The two are still compatible but either the code changes will need to be re-applied or the DB should get cleaned up to remove the changes.

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

snopoke committed:
replace date index with domain, date, id index

snopoke committed:
update queries on MessagingSubevent to use new domain field

snopoke committed:
udpate messaging-event API to sort by date_last_activity by default

this will use the new index and be a more efficient query

millerdev commented on corehq/apps/api/resources/messaging_event/utils.py:
```diff
@@ -39,7 +39,7 @@ def get_order_by_field(request_params):
 
     if not order_by:
-        return "date"
+        return "date_last_activity"
 
     if order_by not in ("date", "-date", "date_last_activity", "-date_last_activity"):
```
Will we need two indexes, one for `date` and another for `date_last_activity`?

snopoke replied:
Ideally yes, but since the DET only uses `date_last_activity` I'm only adding that one now. I'm on the fence about removing option for ordering by `date` entirely.

millerdev replied:
I think we should either add the extra index or remove the option to filter by `date`.

snopoke replied:
I have removed the option to filter by `date`

snopoke committed:
actually call domain backfill function

snopoke committed:
undo changes to use new column except for in the API

The only change currently required is in the API. The other changes
should wait until after the forced back-fill migration has taken place

snopoke committed:
remove sortying by 'date' from messaging-event API

This would require an index on (`domain`, `date`)
to be efficient.

If this is needed in future it can be added.

snopoke commented:
@esoergel @millerdev I have added a few more commits:
* remove the option to filter by `date` since this would require another (large) index and I'm not sure it's required
* undo changes to use the new field (except for in the API). I will re-apply these in PR which adds the auto-migration (https://github.com/dimagi/commcare-hq/pull/32335). My thinking here is to minimize the impact on envs that have not yet done the back-fill (3rd party envs).

snopoke commented:
The data has been back-filled on all Dimagi envs.

esoergel commented:
:+1: on the new commits
